napi-inl.h

```diff
@@ -24,16 +24,21 @@ namespace details {
 template <typename FreeType>
 static inline napi_status AttachData(napi_env env,
                                      napi_value obj,
-                                     FreeType* data) {
+                                     FreeType* data,
+                                     napi_finalize finalizer = nullptr,
+                                     void* hint = nullptr) {
   napi_value symbol, external;
   napi_status status = napi_create_symbol(env, nullptr, &symbol);
   if (status == napi_ok) {
+    if (finalizer == nullptr) {
+      finalizer = [](napi_env /*env*/, void* data, void* /*hint*/) {
+        delete static_cast<FreeType*>(data);
+      };
+    }
     status = napi_create_external(env,
```

Comment:
Seems like we could conditionally replace this with `napi_add_finalizer` if NAPI_VERSION is greater than 4?

Reply:
Yes, we can. I was gonna wait until v8.x drops off, but yeah, I can change it to that.

Reply:
Although I'm not sure. Let's wait until `napi_add_finalizer` is no longer experimental.